src/Task02_PokeAPI/pokeApi.js

```diff
@@ -0,0 +1,172 @@
+      }
+    `,
+    variables: { name: pokemonName.toLowerCase() }
+  }
+
+  try {
+    const response = await fetch('https://graphql-pokeapi.graphcdn.app/', {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json'
+      },
+      body: JSON.stringify(graphqlQuery)
+    })
+
+    const data = await response.json()
+
+    if (data.data && data.data.pokemon) {
+      return data.data.pokemon
+    } else {
+      console.error('No data returned for the specified Pokémon.')
+      return null
+    }
+  } catch (error) {
+    console.error('Error fetching Pokémon data:', error)
+    return null
+  }
+}
+
+function displayPokemon (pokemon) {
+  const container = document.getElementById('pokemon-container')
+
+  const pokemonElement = document.createElement('div')
+  pokemonElement.classList.add('pokemon')
+
+  const nameElement = document.createElement('h2')
+  nameElement.textContent = capitalizeFirstLetter(pokemon.name)
+
+  const imageElement = document.createElement('img')
+  imageElement.src = pokemon.sprites.front_default
+  imageElement.alt = `Sprite of ${pokemon.name}`
+
+  const detailsElement = document.createElement('p')
+  detailsElement.textContent = 'Click for details'
+  detailsElement.classList.add('details-text')
+
+  pokemonElement.appendChild(nameElement)
+  pokemonElement.appendChild(imageElement)
+  pokemonElement.appendChild(detailsElement)
+
+  container.appendChild(pokemonElement)
+
+  pokemonElement.addEventListener('click', () => {
+    showMoreInfo(pokemon)
+  })
+}
+
+function showMoreInfo (pokemon) {
+  document.getElementById('pokemon-name-modal').textContent = capitalizeFirstLetter(pokemon.name)
+  document.getElementById('pokemon-weight-modal').textContent = 'Weight: ' + pokemon.weight
+  document.getElementById('pokemon-height-modal').textContent = 'Height: ' + pokemon.height
+  document.getElementById('pokemon-base-experience-modal').textContent = 'Base experience: ' + pokemon.base_experience
+  document.getElementById('pokemon-image-front-modal').src = pokemon.sprites.front_default
+  document.getElementById('pokemon-image-back-modal').src = pokemon.sprites.back_default
+  document.getElementById('pokemon-types-modal').textContent = 'Types: ' + pokemon.types.map(pt => pt.type.name).join(', ')
+  document.getElementById('pokemon-abilities-modal').textContent = 'Abilities: ' + pokemon.abilities.map(pa => pa.ability.name).join(', ')
+  // document.getElementById('pokemon-stats-modal').textContent = 'Stats: ' + pokemon.stats.map(ps => ps.stat.name).join(', ')
+  document.getElementById('location-area-encounters-modal').textContent = 'Location Area Encounters: ' + pokemon.location_area_encounters
+  document.getElementById('pokemon-modal').style.display = 'block'
+}
+
+function capitalizeFirstLetter (str) {
+  return str.charAt(0).toUpperCase() + str.slice(1)
+}
+
+const pokemonNames = ['Bulbasaur', 'Ivysaur', 'Venusaur', 'Charmander', 'Charmeleon', 'Charizard', 'Squirtle', 'Wartortle']
```

Comment:
you can't hardcode list of pokes like that. you should fetch pokes from API

Reply:
I also noticed, you send request and get details for each poke on page. That is not how it should work. You must send request for details only when user clicks on poke. Don't get details for all pokes, this is not optimal approach